src/dsl/parser.rs

```diff
@@ -181,11 +253,26 @@ impl<'input> Parser<'input> {
             assert_close_prnth = true;
         }
 
-        let rhs = self.parse_rhs_impl()?;
+        let rhs = self.parse_rhs_impl(depth + 1, return_on_dot && !assert_close_prnth)?;
 
         let token = match self.input.peek() {
             Some(token) => token?,
-            None => return Ok(Some((0, Box::new(rhs)))),
+            None => {
+                if rhs.0.len() == 1 {
+                    if let RhsPart::Key(RhsEntry::Key(k)) = rhs.0.get(0).unwrap() {
```

Comment:
unwraps here can be replaced by error propagation

Reply:
this should never panic since we check the length is 1 before calling get